packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -181,17 +181,40 @@ export function useVirtualizer<T extends object, V extends ReactNode, W>(props:
   // Handle scrolling, and call onLoadMore when nearing the bottom.
   let isLoadingRef = useRef(isLoading);
   let prevProps = useRef(props);
-  let onVisibleRectChange = useCallback((rect: Rect) => {
-    state.setVisibleRect(rect);
+  let rectTracker = useRef({
+    // Track the last two visible rect sizes
+    lastTwoRects: [],
+    // Number of times the newest visible rect size has matched the last two sizes
+    sameSizeCounter: 0
+  }).current;
 
+  let onVisibleRectChange = useCallback((rect: Rect) => {
     if (!isLoadingRef.current && onLoadMore) {
       let scrollOffset = state.virtualizer.contentSize.height - rect.height * 2;
       if (rect.y > scrollOffset) {
         isLoadingRef.current = true;
         onLoadMore();
       }
     }
-  }, [onLoadMore, state]);
+
+    if (rectTracker.lastTwoRects.length > 1) {
+      if (rectTracker.lastTwoRects.some(oldRect => rect.equals(oldRect))) {
+        ++rectTracker.sameSizeCounter;
+        if (rectTracker.sameSizeCounter > 5) {
+          return;
```

Comment:
Should it get reset before early returning? Just thinking what would happen if another real resize happens.

Reply:
If onVisibleRectChange is triggered by a real resize and returns a `rect` that matches one of the two old rects I don't think we wanna reset the counter since we can't really distinguish that from the failure case right? I figure a user causing a resize of the virtualizer will typically hit a rect that is different from the last two rects and thus it will reset the counter then